client/src/components/LayerPlayer.jsx

```diff
@@ -27,33 +27,44 @@ export default function LayerPlayer({ layers, trackId, userId, recordingHandler,
   const playAllLayers = async () => {
     try {
       await Tone.start();
+      await Tone.loaded();
+
       let keys = Object.keys(playerStore.allPlayers)
       keys.forEach((layerKey, i) => {
-        let layer = playerStore.allPlayers[layerKey]
-        layer.start()
+
         Tone.Transport.schedule((time) => {
           Tone.Draw.schedule(() => {
             renderWaveform(layer.waveform, layerKey);
           }, time);
         }, "+0.005");
+        let layer = playerStore.allPlayers[layerKey]
+        console.log('LAYER',layer.trimFromStart)
+        layer.start((layer.trimFromStart),layer.trimFromStart,layer.layerData.duration)
```

Comment:
We can probably change layer.layerData.duration to just layer.duration() just to make sure we have the most up to date value from the player buffer.